ipfsconn/ipfshttp/ipfshttp.go

```diff
@@ -450,6 +459,97 @@ func checkResponse(path string, code int, body []byte) error {
 	return fmt.Errorf("IPFS-post '%s' unsuccessful: %d: %s", path, code, body)
 }
 
+// postCtxStreaming makes a POST request against
+// the ipfs daemon, reads the body of the response line by line
+// at evey millisecond. It cancels the context if timeout duration
+// has passed since last new data on the buffer.
+func (ipfs *Connector) postCtxStreaming(ctx context.Context, path string, timeout time.Duration) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	timer := time.NewTimer(timeout)
+	defer timer.Stop()
+
+	elapsed := time.Now()
+
+	go func() {
+		for {
+			select {
+			case <-timer.C:
+				{
+					if time.Since(elapsed) >= timeout {
+						cancel()
+						break
```

Comment:
this is just breaking from the select, not from the for loop?

Reply:
Yes, but once context is canceled our `reader` would throw an error `context cancelled` and the function would return.
I could break out of the loop by keeping some done flag, but that would add a bunch of lines of redundant code.

Reply:
Unresolve if you think I should explicitly break out of this for loop and other infinite loops

Reply:
Your reader will return and the other function will end but this goroutine stays there forever (?)

Reply:
Which makes me realize that this goroutine is also not cleaned up when things don't time out.